src/sidebar/components/markdown-editor.js

```diff
@@ -147,9 +165,14 @@ ToolbarButton.propTypes = {
 };
 
 /**
- * @typedef {string} ButtonID
- * A unique string that can be one of one of:
- * [bold, italic, quote, link, image, math, numlist, list, preview, help]
+ * @typedef {'bold'|'italic'|'quote'|'link'|'image'|'math'|'numlist'|'list'|'preview'|'help'} ButtonID
+ */
+
+/**
+ * @typedef ToolbarProps
+ *  @prop {boolean} [isPreviewing] - `true` if the editor's "Preview" mode is active.
+ *  @prop {(a: ButtonID) => any} [onCommand] - Callback invoked with the selected command when a toolbar button is clicked.
+ *  @prop {() => any} [onTogglePreview] - Callback invoked when the "Preview" toggle button is clicked.
```

Comment:
I expected to see a boolean argument to `onTogglePreview` here, but looking at the code this type is correct. It is the caller's job to figure out whether the preview is being turned on or off.